packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/stage-parser.ts

```diff
@@ -18,32 +18,72 @@ export function setNodeDisabled(node: t.Node, value: boolean) {
   (node as any)[kDisabled] = value;
 }
 
-export function isStageLike(node?: t.Node | null, loose = false): node is StageLike {
-  return (
-    !!node &&
-    node.type === 'ObjectExpression' &&
-    node.properties.length === 1 &&
-    node.properties[0].type === 'ObjectProperty' &&
-    node.properties[0].key &&
-    ((node.properties[0].key.type === 'Identifier' &&
-      node.properties[0].key.name.startsWith('$')) ||
-      (node.properties[0].key.type === 'StringLiteral' &&
-        node.properties[0].key.value.startsWith('$'))) &&
-    (loose || node.properties[0].value != null)
-  );
+export function isValidStageNode(node?: t.ObjectExpression): boolean {
+  if (node) {
+    try {
+      // TODO: either export ejson-shell-parser logic that validates the AST or
+      // move the logic in this package to avoid generating source from ast
+      // before validating it again
+      parseEJSON(generate(node));
```

Comment:
Can we rename this? I know the `ejson-shell-parser` name is bad, but I’m pretty sure naïvely looking at this code one would assume that `parseEJSON` parses EJSON, when in reality it parses something else entirely

Reply:
Yeah, can do 👍 Something like `parseShellJSON` or `parseBSON`? Do you have a preference?

Reply:
`parseShellBSON`? JSON and BSON are both (other) different standalone formats, but “Shell BSON” is a terminology we use in other places so that might fit

Reply:
Yep, makes sense, will go with that! (I always forget whether we call it shell bson or shell json, maybe it will finally stick 😄)